cpp/scenario/controllers/src/ComputedTorqueFixedBase.cpp

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This project is dual licensed under LGPL v2.1+ or Apache License.
+ *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "scenario/controllers/ComputedTorqueFixedBase.h"
+#include "scenario/controllers/References.h"
+#include "scenario/gazebo/Joint.h"
+#include "scenario/gazebo/Log.h"
+#include "scenario/gazebo/Model.h"
+
+#include <Eigen/Dense>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/FreeFloatingState.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+
+#include <cassert>
+#include <unordered_map>
+
+using namespace scenario::controllers;
+
+class ComputedTorqueFixedBase::Impl
+{
+public:
+    class Buffers;
+
+    std::string urdfFile;
+
+    struct
+    {
+        std::vector<double> kp;
+        std::vector<double> kd;
+        std::array<double, 3> gravity;
+        std::unordered_map<std::string, base::JointControlMode> controlMode;
+    } initialValues;
+
+    JointReferences jointReferences;
+    std::unique_ptr<Buffers> buffers;
+    std::unique_ptr<iDynTree::KinDynComputations> kinDyn;
+
+    static Eigen::Map<Eigen::VectorXd>
+    toEigen(const std::vector<double>& vector)
```

Comment:
Since the content of the vector can be changed through the Eigen map, maybe passing a non-const reference would be more correct. And we also would avoid the cost cast.

Reply:
Fixed in 48eca76a0946e9c7739dc013317490dc57a8a4ea